test/testdata/lsp/hover_method_includes_defs.rb

```diff
@@ -0,0 +1,138 @@
+
+  sig {returns T.untyped}
+  def module_usages
+    s = "Foo"
+    [
+      module_method(s),
+      # ^ hover: def module_method(name)
+      OuterModule.module_self_method(s),
+      #           ^ hover: def self.module_self_method(name)
+      OuterModule::module_self_method(s),
+      #            ^ hover: def self.module_self_method(name)
+    ]
+  end
+
+  class InnerClass
+    extend T::Sig
+    sig {params(name: String).void}
+    def initialize(name)
+      @name = name
+    end
+
+    sig {params(name: String).returns(Integer)}
+    def self.class_self_method(name)
+      name.length
+    end
+  
+    sig {void}
+    def no_args_return_void
+      # no args
+    end
+
+    sig {returns(Integer)}
+    def no_args_return_value
+      "no args".length
+    end
+
+    sig {params(fname: String, lname: String).returns(String)}
+    def positional_args(fname, lname)
+      "#{fname}:#{lname}"
+    end
+
+    sig {params(fname: String, lname: String).returns(String)}
+    def keyword_args_no_defaults(fname:, lname:)
+      "#{fname}:#{lname}"
+    end
+
+    sig {params(fname: String, lname: String).returns(String)}
+    def keyword_args_with_defaults(fname: "Jane", lname: "Doe")
+      "#{fname}:#{lname}"
+    end
+
+    sig {params(names: String).returns(String)}
+    def splat_args(*names)
+      names.join ':'
+    end
+
+    sig {params(blk: T.proc.params(arg0: Integer).returns(String)).returns(String)}
+    def block_args(&blk)
+      blk.call(1)
+    end
+
+    sig {params(pos: String, splat: String, required_key: String, optional_key: String, block_arg: T.proc.params(a: String).returns(String)).returns(String)}
+    def multiple_arg_types(pos, *splat, required_key:, optional_key: "Jane", &block_arg)
+      "#{pos} #{required_key} #{optional_key} #{splat.join ':'} #{block_arg.call(pos)}"
+    end
+
+    sig {returns T.untyped}
+    def class_usages
+      s = "Foo"
+      qualified = InnerClass.new(s)
+                             # ^ hover: def initialize(name)
+      [
+        no_args_return_void,
+        # ^ hover: def no_args_return_void
+        no_args_return_value,
+        # ^ hover: def no_args_return_value
+        positional_args(s, s),
+        # ^ hover: def positional_args(fname, lname)
+        keyword_args_no_defaults(fname: s, lname: s),
+        # ^ hover: def keyword_args_no_defaults(fname:, lname:)
+        keyword_args_with_defaults,
+        # ^ hover: def keyword_args_with_defaults(fname:…, lname:…)
+        splat_args,
+        # ^ hover: def splat_args(*names)
+        splat_args(s, s, s, s, s),
+        # ^ hover: def splat_args(*names)
+        block_args {|x| "blk#{x}"},
+        # ^ hover: def block_args(&blk)
+        multiple_arg_types(s, s, s, s, required_key: s) {|x| x},
+        # ^ hover: def multiple_arg_types(
+        # ^ hover:   pos,
+        # ^ hover:   *splat,
+        # ^ hover:   required_key:,
+        # ^ hover:   optional_key:…,
+        # ^ hover:   &block_arg
+        # ^ hover: )
```

Comment:
Cool nice to see that multi line assertions work like this!

We might want some way to assert they’re in the right order though 🤔

Maybe in a future change

Reply:
To add to this: The hover assertion merely asserts that the contents match a specific line. We don't want them to match every line, since we put every sig+def into a markdown codeblock.

Relative ordering might be a good future modification!